src/Microsoft.AspNet.Mvc.Core/Rendering/Html/DefaultHtmlGenerator.cs

```diff
@@ -1024,6 +1024,16 @@ internal static object GetModelStateValue(ViewContext viewContext, string key, T
                         attributeValue = useViewData ? EvalString(viewContext, fullName, format) : valueParameter;
                     }
 
+                    object typeAttributeValue = null;
+                    if (htmlAttributes != null && htmlAttributes.TryGetValue("type", out typeAttributeValue))
+                    {
+                        if (string.Equals(typeAttributeValue.ToString(), "file") ||
+                            string.Equals(typeAttributeValue.ToString(), "image"))
```

Comment:
add `StringComparison.OrdinalIgnoreCase`.  this comparison should not be case-sensitive (and we are always explicit about comparison types in product code).